gradoop-flink/src/test/java/org/gradoop/flink/model/impl/operators/kmeans/KMeansTest.java

```diff
@@ -0,0 +1,173 @@
+    expectedNearestCenter.add(new Tuple2<>(new Centroid(3, new Point(25.0, 30.0)), new Point(35.0, 25.0)));
+    expectedNearestCenter.add(new Tuple2<>(new Centroid(3, new Point(25.0, 30.0)), new Point(40.0, 30.0)));
+    expectedNearestCenter.add(new Tuple2<>(new Centroid(1, new Point(10.0, 15.0)), new Point(10.0, 10.0)));
+    expectedNearestCenter.add(new Tuple2<>(new Centroid(2, new Point(20.0, 20.0)), new Point(10.0, 30.0)));
+    expectedNearestCenter.add(new Tuple2<>(new Centroid(3, new Point(25.0, 30.0)), new Point(20.0, 40.0)));
+    expectedNearestCenter.add(new Tuple2<>(new Centroid(1, new Point(10.0, 15.0)), new Point(5.0, 10.0)));
+
+
+    expectedNearestCenterWithAppendedCounter.add(new Tuple3<>(2, new Point(20.0, 10.0), 1L));
+    expectedNearestCenterWithAppendedCounter.add(new Tuple3<>(2, new Point(25.0, 15.0), 1L));
+    expectedNearestCenterWithAppendedCounter.add(new Tuple3<>(2, new Point(30.0, 20.0), 1L));
+    expectedNearestCenterWithAppendedCounter.add(new Tuple3<>(3, new Point(35.0, 25.0), 1L));
+    expectedNearestCenterWithAppendedCounter.add(new Tuple3<>(3, new Point(40.0, 30.0), 1L));
+    expectedNearestCenterWithAppendedCounter.add(new Tuple3<>(1, new Point(10.0, 10.0), 1L));
+    expectedNearestCenterWithAppendedCounter.add(new Tuple3<>(2, new Point(10.0, 30.0), 1L));
+    expectedNearestCenterWithAppendedCounter.add(new Tuple3<>(3, new Point(20.0, 40.0), 1L));
+    expectedNearestCenterWithAppendedCounter.add(new Tuple3<>(1, new Point(5.0, 10.0), 1L));
+
+    expectedCentroidsWithAccumulatedPoints.add(new Tuple3<>(1, new Point(15.0, 20.0), 2L));
+    expectedCentroidsWithAccumulatedPoints.add(new Tuple3<>(2, new Point(85.0, 75.0), 4L));
+    expectedCentroidsWithAccumulatedPoints.add(new Tuple3<>(3, new Point(95.0, 95.0), 3L));
+
+    expectedAverageCentroids.add(new Centroid(1, new Point(7.5, 10.0)));
+    expectedAverageCentroids.add(new Centroid(2, new Point(21.25, 18.75)));
+    expectedAverageCentroids.add(new Centroid(3, new Point(31.666666666666668, 31.666666666666668)));
+
+    inputGdlGraph =
+      "g[" + "(c1:Coordinate {lat: 10.0, long: 15.0})" + "(c2:Coordinate {lat: 20.0, long: 20.0})" +
+        "(c3:Coordinate {lat: 25.012F, long: 30.012F})" + "(c4:Coordinate {lat: 20L, long: 10L})" +
+        "(c5:Coordinate {lat: 25, long: 15})" + "(c6:Coordinate {lat: 30.0, long: 20.0})" +
+        "(c7:Coordinate {lat: 35.0, long: 25.0})" + "(c8:Coordinate {lat: 40.0, long: 30.0})" +
+        "(c9:Coordinate {lat: 10.0, long: 10.0})" + "(c10:Coordinate {lat: 10.0, long: 30.0})" +
+        "(c11:Coordinate {lat: 20.0, long: 40.0})" + "(c12:Coordinate {lat: 5.0, long: 10.0})" + "]";
+  }
+
+
+  //Unit Tests
+  @Test
+  public void testSelectNearestCenter() throws Exception {
+
+    List<Tuple2<Centroid, Point>> nearestCenter =
+      points.map(new SelectNearestCenter()).withBroadcastSet(centroids, "centroids").collect();
```

Comment:
This is also for your other test functions - try to avoid .collect() in test functions. Better create an empty list that you can fill with LocalCollectionOutputFormat (see other test functions). Do not forget calling `getExecutionEnvironment().execute(); ` to fill the array.

Reply:
Done